leakcanary-android-core/src/main/java/leakcanary/internal/activity/LeakActivity.kt

```diff
@@ -60,6 +60,18 @@ internal class LeakActivity : NavigatingActivity() {
     leaksButton.setOnClickListener { resetTo(LeaksScreen()) }
     heapDumpsButton.setOnClickListener { resetTo(HeapDumpsScreen()) }
     aboutButton.setOnClickListener { resetTo(AboutScreen()) }
+
+    handleViewHprof(intent)
+  }
+
+  private fun handleViewHprof(intent: Intent?) {
+    if (intent?.action != Intent.ACTION_VIEW) return
+    val uri = intent.data ?: return
+    if (uri.lastPathSegment?.endsWith(".hprof") != true) return
```

Comment:
Worth displaying a toast to surface the error?

Reply:
Much better, will do.